indico/web/client/js/custom_elements/ind_date_picker.js

```diff
@@ -479,9 +479,18 @@ function toOptionalDate(dateString) {
 
 // Localization
 
+// This is used in browsers that do not support Intl API.
+// Only those items where the data differs from the generic
+// fallback are listed.
+const FALLBACK_WEEK_INFO = {
+  'en-US': {firstDay: 7, weekend: [6, 7]},
+  'pt-BR': {firstDay: 7, weekend: [6, 7]},
+};
+const GENERIC_FALLBACK_WEEK_INFO = {firstDay: 1, weekend: [6, 7]};
```

Comment:
Could we get this information from moment? I want to avoid having yet another place that needs manual updating when we add a new translation...

Reply:
Moment is pretty much unmaintained, so it's basically the same as leaving this in forever, so I think that would be a micro optimization that has no benefit.

Reply:
On the other hand, I could look into generating this from NodeJS at build time. That might be a more worthwhile investment of our time.

Reply:
Maybe simply adding this polyfill would do the job: https://github.com/bart-krakowski/get-week-info-polyfill/tree/main

Reply:
I gave it a quick try and this works fine:

```patch
diff --git a/indico/web/client/js/custom_elements/ind_date_picker.js b/indico/web/client/js/custom_elements/ind_date_picker.js
index d1a335132d..f2ec4826b0 100644
--- a/indico/web/client/js/custom_elements/ind_date_picker.js
+++ b/indico/web/client/js/custom_elements/ind_date_picker.js
@@ -10,6 +10,7 @@ import {formatDate} from 'indico/utils/date_format';
 import {createDateParser} from 'indico/utils/date_parser';
 import {topBottomPosition} from 'indico/utils/positioning';

+import '@bart-krakowski/get-week-info-polyfill';
 import './ind_date_picker.scss';

 const DEFAULT_LOCALE = document.documentElement.dataset.canonicalLocale;
@@ -479,9 +480,9 @@ function toOptionalDate(dateString) {

 // Localization

-function getWeekInfoForLocale(locale) {
-  let weekInfo = new Intl.Locale(locale).weekInfo;
-  weekInfo ??= {firstDay: 7, weekend: [6, 7]}; /* Firefox 😭 */
+function getWeekInfoForLocale(localeName) {
+  const locale = new Intl.Locale(localeName)
+  const weekInfo = locale.weekInfo ?? locale.getWeekInfo?.();
   weekInfo.weekDays = [];
   for (let i = weekInfo.firstDay, end = i + 7; i < end; i++) {
     weekInfo.weekDays.push(i % 7 || 7);
diff --git a/package-lock.json b/package-lock.json
index 1f3fbf967d..968a30575b 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -10,6 +10,7 @@
       "license": "MIT",
       "dependencies": {
         "@babel/runtime": "^7.18.3",
+        "@bart-krakowski/get-week-info-polyfill": "^1.0.8",
         "@dr.pogodin/babel-plugin-react-css-modules": "~6.1.1",
         "@monaco-editor/react": "^4.6.0",
         "@tinymce/tinymce-react": "^4.3.0",
@@ -2216,6 +2217,11 @@
         "node": ">=6.9.0"
       }
     },
+    "node_modules/@bart-krakowski/get-week-info-polyfill": {
+      "version": "1.0.8",
+      "resolved": "https://registry.npmjs.org/@bart-krakowski/get-week-info-polyfill/-/get-week-info-polyfill-1.0.8.tgz",
+      "integrity": "sha512-Vkp5Skl44AkQ/w+pJ1XrCa5P5whvpK7Ix75KBoZXpGXOEp6CqLVkcAQaWfzY6+rF1R+2cwvJMkxkYF3M4B8HFw=="
+    },
     "node_modules/@bcoe/v8-coverage": {
       "version": "0.2.3",
       "resolved": "https://registry.npmjs.org/@bcoe/v8-coverage/-/v8-coverage-0.2.3.tgz",
diff --git a/package.json b/package.json
index ceff24330c..b4a2db03dd 100644
--- a/package.json
+++ b/package.json
@@ -8,6 +8,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.18.3",
+    "@bart-krakowski/get-week-info-polyfill": "^1.0.8",
     "@dr.pogodin/babel-plugin-react-css-modules": "~6.1.1",
     "@monaco-editor/react": "^4.6.0",
     "@tinymce/tinymce-react": "^4.3.0",
```

Reply:
I don't understand what the point is. The guy just did the same thing I did, and the only difference is the package provides a lot more code for cases we won't have.

Reply:
That was my response to "On the other hand, I could look into generating this from NodeJS at build time. That might be a more worthwhile investment of our time." since it seems a bit pointless to do the same work again.

But if you want to generate it at build time just for the locales we have that's fine for me as well. We already have support for the `preval` babel macro to do generate such things at build time w/o any custom build scripts etc. (check .../outdatedbrowser/index.js for an example).

Reply:
> since it seems a bit pointless to do the same work again.

Well, it depends on what we're doing. If our goal is to keep this up to date, then it makes sense to generate it at build time from an up-to-date version of NodeJS.

<img width="296" alt="image" src="https://github.com/user-attachments/assets/a98093e6-f7a9-4e59-8c75-e393011cbe23">

This can go into a define plugin to be defined as a global and used either as the authoritative value or as a fallback.